lib/src/json_schema/json_schema.dart

```diff
@@ -549,8 +549,11 @@ class JsonSchema {
   }
 
   /// Look for the given anchor at the schema. Returns null if nothing is found.
-  JsonSchema _resolveDynamicAnchor(JsonSchema schema, String dynamicAnchor) {
-    if (this.schemaVersion < SchemaVersion.draft2020_12) {
+  JsonSchema _resolveDynamicAnchor(String dynamicAnchor, JsonSchema schema) {
```

Comment:
more idiomatic to do `JsonSchema _resolveDynamicAnchor(String dynamicAnchor, {JsonSchema schema = this})`